### PR TITLE
[boost] Build boost with zlib support

### PR DESCRIPTION
**Pull request by mmassing**
This PR builds boost with support for zlib compression/decompression in the iostreams library.

### CODE REVIEW

File: ports/boost/portfile.cmake [hunk 1]
```diff
@@ -31,6 +31,10 @@ endif()
 message(STATUS "Bootstrapping done")
 
 set(B2_OPTIONS
+    -sZLIB_BINARY=zlib
+    -sZLIB_INCLUDE="${CURRENT_INSTALLED_DIR}\\include"
+    -sZLIB_LIBPATH="${CURRENT_INSTALLED_DIR}\\lib"
```
Comment (ras0219-msft) [inline ports/boost/portfile.cmake]: This setting needs to be different in Release and Debug -- for Debug you should point it at `${CURRENT_INSTALLED_DIR}\\debug\\lib`.

Comment (mmassing) [inline ports/boost/portfile.cmake]: Thanks for the correction, I've now used different paths for Release and Debug. 

BTW, I've already signed the CLA, will the label be automatically updated?

**Comment by ras0219-msft** [comment]
@mmassing It looks like the CLA bot is working for new PRs; could you re-submit this with the fix above as a new PR? Thanks!

**Comment by rjoco** [comment]
ZLib support in boost.iostreams appears to be no longer supported in the latest revision. References to ZLIB were removed from ports/boost/portfile.cmake and the equivalent code doesn't seem to be in boost-modular.cmake...

**Pull request closed without merging**